### PR TITLE
[CHK-2370] feat: added new command for handle logic for execute async close payment

### PR DESCRIPTION
**Pull request by Fernando-Granato**
depend on https://github.com/pagopa/pagopa-ecommerce-local/pull/130
depend on https://github.com/pagopa/pagopa-ecommerce-commons/pull/169

#### List of Changes
Modified the flow that performed close payment in case of v2 type events. Specifically, the flow now involves writing the new async `CLOSURE_REQUESTED_EVENT` event. 
The system will take care of changing the state of the view to `CLOSURE_REQUESTED`
Created the new configuration UniqueIdUtils class to initialize the Bean uniqueIdUtils since the Component annotation on the class present in the commons library was removed

#### Motivation and Context
The change was made to be able to handle asynchronous close payment in case of v2 type events.

#### How Has This Been Tested?
Unit tests were written and performed some testing with ecommerce local

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by infantesimone — COMMENTED** [review]
sounds good, I'll do another review tomorrow :-)

File: src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionSendClosureRequestedHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+                )
+                .verifyComplete();
+
+        verify(transactionEventClosureRequestedRepository, times(1)).save(any());
+        verify(transactionSendClosureRequestQueueClient, times(1)).sendMessageWithResponse(any(), any(), any());
+        assertEquals(Duration.ofSeconds(transientQueueEventsTtlSeconds), durationCaptor.getValue());
+    }
+
+    @Test
+    void shouldSaveClosureRequestedEventWithErrorTransactionNotFound() {
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionSendClosureRequestedHandlerTest.java:122]:
```suggestion
    void shouldNotSaveClosureRequestedEventWithErrorTransactionNotFound() {
```

File: src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionSendClosureRequestedHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        StepVerifier.create(transactionSendClosureRequestHandler.handle(transactionClosureRequestCommand))
+                .expectError(TransactionNotFoundException.class)
+                .verify();
+
+        verify(transactionEventClosureRequestedRepository, times(0)).save(any());
+        verify(transactionSendClosureRequestQueueClient, times(0)).sendMessageWithResponse(any(), any(), any());
+    }
+
+    @Test
+    void shouldSaveClosureRequestedEventWithErrorAlreadyProcessedException() {
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionSendClosureRequestedHandlerTest.java:143]:
```suggestion
    void shouldNotSaveClosureRequestedEventWithErrorAlreadyProcessedException() {
```

File: api-spec/v1/transactions-api.yaml [hunk 3]
```diff
@@ -742,6 +742,7 @@ components:
         - ACTIVATED
         - AUTHORIZATION_REQUESTED
         - AUTHORIZATION_COMPLETED
+        - CLOSURE_REQUESTED
```
Comment (pietro-tota) [inline api-spec/v1/transactions-api.yaml:745]: TODO: remember to update this specs also on infra

**Review by giovanniberti — APPROVED** [review]

File: src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionSendClosureRequestedHandlerTest.java (hunk 1 above)
Comment (Fernando-Granato) [reply src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionSendClosureRequestedHandlerTest.java]: Fixed with 6bba981ca77701780ef8bff155faaa209d8e3d13

File: src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionSendClosureRequestedHandlerTest.java (hunk 2 above)
Comment (Fernando-Granato) [reply src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionSendClosureRequestedHandlerTest.java]: Fixed with 6bba981ca77701780ef8bff155faaa209d8e3d13

File: api-spec/v1/transactions-api.yaml (hunk 3 above)
Comment (Fernando-Granato) [reply api-spec/v1/transactions-api.yaml:745]: https://github.com/pagopa/pagopa-infra/pull/1686

**Comment by infantesimone** [comment]
ping :-)

**Review by pietro-tota — DISMISSED** [review]
Well done 🚀  just some trivial issues

File: src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionSendClosureRequestedHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+@ExtendWith(MockitoExtension.class)
+class TransactionSendClosureRequestedHandlerTest {
+
+    private it.pagopa.transactions.commands.handlers.v2.TransactionSendClosureRequestHandler transactionSendClosureRequestHandler;
+
+    @Mock
+    private TransactionsEventStoreRepository<Void> transactionEventClosureRequestedRepository;
+
+    @Mock
+    private QueueAsyncClient transactionSendClosureRequestQueueClient;
```
[the remark this answers is not in the document]
Comment (Fernando-Granato) [reply src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionSendClosureRequestedHandlerTest.java]: Fixed with c5f03305484b56eb366394d228ccc6f31c4bbd87

File: src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionSendClosureRequestedHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class TransactionSendClosureRequestedHandlerTest {
+
+    private it.pagopa.transactions.commands.handlers.v2.TransactionSendClosureRequestHandler transactionSendClosureRequestHandler;
+
+    @Mock
+    private TransactionsEventStoreRepository<Void> transactionEventClosureRequestedRepository;
```
[the remark this answers is not in the document]
Comment (Fernando-Granato) [reply src/test/java/it/pagopa/transactions/commands/handlers/v2/TransactionSendClosureRequestedHandlerTest.java]: Fixed with c5f03305484b56eb366394d228ccc6f31c4bbd87

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**